### PR TITLE
Legend portal

### PR DESCRIPTION
**Pull request by ckifer**
## Description

Render Legend in the `recharts-wrapper` container so that the Legend `foreignObject` does not overlay on top of everything and disrupt mouse events

## Related Issue

3.0 related issues but not a specific one

## Motivation and Context

- adds a new feature
- prevents disabling tooltip for charts that have `item` tooltipType when legend is also rendered

## How Has This Been Tested?

- unit tests passing
- additional tests added for legend portal
- new storybook

Visual diff
https://www.chromatic.com/build?appId=63da8268a0da9970db6992aa&number=1374

Seems that icons are now better aligned with their text

## Screenshots (if appropriate):

image

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

I didn't actually add a storybook entry for legend portal yet, I can do that later

- [X] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [X] I have added a storybook story or extended an existing story to show my changes

### CODE REVIEW

File: storybook/stories/API/component/Legend.stories.tsx [hunk 1]
```diff
@@ -36,3 +36,66 @@ export const API = {
...
+                bottom: 20,
+                left: 20,
+              }}
+              data={rechartsPackageDownloads}
+            >
+              {/* The target component */}
+              {legendPortal && (
+                <Legend
+                  {...args}
+                  // TODO: when a portal is passed remove absolute positioning and other styles
```
Comment (PavelVanecek) [inline storybook/stories/API/component/Legend.stories.tsx:74]: When should we fix this TODO? In this PR? Or before releasing 3.x?

File: test/component/Legend.spec.tsx [hunk 2]
```diff
@@ -2930,6 +2992,49 @@ describe('<Legend />', () => {
...
+          <Legend />
+          <Scatter dataKey="percent" />
+        </ScatterChart>,
+      );
+
+      expect(container.querySelectorAll('.recharts-wrapper svg .recharts-legend-wrapper')).toHaveLength(0);
+      expect(container.querySelector('.recharts-wrapper > .recharts-legend-wrapper')).toBeVisible();
+    });
+
+    it('should render in a custom portal if "portal" prop is set', () => {
```
Comment (PavelVanecek) [inline test/component/Legend.spec.tsx:3008]: Do you think it's valuable to test what happens when `portal` prop is undefined? Or some other invalid value?

**Review by PavelVanecek — APPROVED** [review]

File: storybook/stories/API/component/Legend.stories.tsx (hunk 1 above)
Comment (ckifer) [reply storybook/stories/API/component/Legend.stories.tsx:74]: Didn't want to do it in this PR - at this point it's at the same place as Tooltip. I can take both Tooltip and this next thing I do

File: test/component/Legend.spec.tsx (hunk 2 above)
Comment (ckifer) [reply test/component/Legend.spec.tsx:3008]: Yeah sure, I can add something. Although if it's undefined it uses the one from the chart

Comment (ckifer) [reply test/component/Legend.spec.tsx:3020]: done

**Pull request merged**